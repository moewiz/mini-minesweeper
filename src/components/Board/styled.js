import styled from "styled-components";

export const BoardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
