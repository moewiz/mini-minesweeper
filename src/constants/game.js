export const BEGINNER = {
  size: 9,
  mines: 10
};

export const ADVANTAGE = {
  size: 16,
  mines: 40
};

export const GAME_STATUS = {
  LOSE: -1,
  NOT_START: 0,
  PLAYING: 1,
  WIN: 2
};
