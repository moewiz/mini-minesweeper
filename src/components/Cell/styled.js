import styled, { css } from "styled-components";

export const CellWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border: 1px solid;
  border-color: #000;
  background: ${props => (props.isOpen ? "#e4e4e4" : "#fff")};
  font-weight: 600;
  ${props =>
    props.isMine &&
    props.isOpen &&
    css`
      position: relative;
      &:after {
        content: "";
        position: absolute;
        background: #24292e !important;
        border: 1px solid #24292e;
        border-radius: 50%;
        top: 4px;
        left: 4px;
        width: 20px;
        height: 20px;
      }
    `}
  &:hover {
    background: #e4e4e4;
  }
`;
