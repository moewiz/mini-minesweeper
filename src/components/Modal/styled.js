import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const ButtonWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const HeaderBtn = css`
  color: #000;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export const NewGameStyled = styled.div`
  ${HeaderBtn}
`;

export const HomeLinkStyled = styled(Link)`
  ${HeaderBtn}
`;

export const MessageStyled = styled.div`
  font-size: 20px;
  text-align: center;
  margin-top: 5px;
`;
