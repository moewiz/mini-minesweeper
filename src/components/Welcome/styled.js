import styled from "styled-components";
import { Link } from "react-router-dom";

export const HomeWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const GroupLevelStyled = styled.div`
  display: flex;
  flex-direction: row;
  width: 350px;
  justify-content: space-around;
`;

export const LevelBtnStyled = styled(Link)`
  font-size: 20px;
  color: #000;
  border: 1px solid;
  padding: 15px;
  text-decoration: none;
  display: flex;
  &:hover {
    text-decoration: underline;
  }
`;
