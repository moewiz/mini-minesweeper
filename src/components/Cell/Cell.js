import React from "react";
import PropTypes from "prop-types";
import { CellWrapper } from "./styled";
import { GAME_STATUS } from "../../constants/game";

const Cell = ({ cell, openCell, gameStatus }) => {
  return (
    <CellWrapper
      isOpen={
        cell.isOpen ||
        gameStatus === GAME_STATUS.WIN ||
        gameStatus === GAME_STATUS.LOSE
      }
      isMine={cell.minesAround === -1}
      onClick={() => openCell(cell)}
    >
      {cell.isOpen && cell.minesAround > 0 && cell.minesAround}
    </CellWrapper>
  );
};

Cell.propTypes = {
  cell: PropTypes.shape({
    minesAround: PropTypes.number.isRequired,
    isOpen: PropTypes.bool.isRequired,
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired
  }),
  openCell: PropTypes.func.isRequired,
  gameStatus: PropTypes.number.isRequired
};

export default Cell;
