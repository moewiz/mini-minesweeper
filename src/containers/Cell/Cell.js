import React from "react";
import { Cell } from "../../components";
import { GAME_STATUS } from "../../constants/game";

const CellContainer = props => {
  const openCell = cell => {
    console.log("click on cell", cell);
  };
  const gameStatus = GAME_STATUS.WIN;

  return <Cell {...props} openCell={openCell} gameStatus={gameStatus} />;
};

export default CellContainer;
