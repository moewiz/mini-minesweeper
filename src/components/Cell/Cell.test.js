import React from "react";
import { shallow, mount } from "enzyme";
import Cell from "./Cell";
import { CellWrapper } from "./styled";
import { GAME_STATUS } from "../../constants/game";

describe("<Cell />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const openCellMock = jest.fn();
  const CellComponent = (
    <Cell
      cell={{
        isOpen: false,
        x: 0,
        y: 0,
        minesAround: 0
      }}
      openCell={openCellMock}
      gameStatus={GAME_STATUS.NOT_START}
    />
  );

  it("should render without throwing an error", () => {
    expect(shallow(CellComponent).length).toBe(1);
  });
  it("should match snapshot", () => {
    expect(shallow(CellComponent)).toMatchSnapshot();
  });
  it("should call openCellMock 1 time when clicked", () => {
    const CellWrapperElement = mount(CellComponent)
      .find(CellWrapper)
      .at(0);
    CellWrapperElement.simulate("click");

    expect(openCellMock).toBeCalledTimes(1);
  });
});
