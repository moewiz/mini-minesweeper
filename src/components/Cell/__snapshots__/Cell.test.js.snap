// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should match snapshot 1`] = `
<styled.div
  isMine={false}
  isOpen={false}
  onClick={[Function]}
/>
`;
